### DISCUSSION TITLE
Shelly Plus 1 random "Reset to Defaults"

### DISCUSSION THREAD

**Discussion by 1Gootch1** [arendst/Tasmota#14795] [Q&A] [answered]
### PROBLEM DESCRIPTION

I successfully flashed [tasmota32solo1.bin](http://ota.tasmota.com/tasmota32/release/tasmota32solo1.bin) to my Shelly Plus 1.
I also tried the latest development release.
But after performing the configuration the Shelly performe randomly resets to default.
I think it is related to configuring the GPIO 4 as Button

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues) might be related to #10237
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Shelly Plus 1
- [x] Tasmota binary firmware version number used: tasmota32solo1.bin
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: esptool
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE

I think it is related to configure GPIO4 as Button.
I tried it on two different Shelly Plus 1 and it is the same behavior on both.

### EXPECTED BEHAVIOUR
don't performe random resets

### ADDITIONAL CONTEXT
This is the serial output Log Level set to 4

```lua
 14:13:48.009 WIF: Checking connection...
14:13:50.646 SRC: Button
14:13:50.647 CMD: Grp 0, Cmnd 'RESET', Idx 1, Len 1, Data '1'
14:13:50.650 RSL: RESULT = {"Reset":"Reset and Restarting"}
14:13:50.748 CFG: Use defaults
14:13:50.750 CFG: CR 135/699, Busy 0
14:13:50.750 CFG: CR 142/699, Busy 0
14:13:50.751 CFG: CR 150/699, Busy 0
14:13:50.751 CFG: CR 158/699, Busy 0
14:13:50.765 CFG: CR 166/699, Busy 0
14:13:50.765 CFG: CR 173/699, Busy 0
14:13:50.766 CFG: CR 232/699, Busy 0
14:13:50.767 CFG: CR 232/699, Busy 0
14:13:50.777 CFG: CR 232/699, Busy 0
14:13:50.778 CFG: CR 232/699, Busy 0
14:13:50.778 CFG: CR 232/699, Busy 0
14:13:50.779 CFG: CR 239/699, Busy 0
14:13:50.790 CFG: CR 239/699, Busy 0
14:13:50.790 CFG: CR 239/699, Busy 0
14:13:50.791 CFG: CR 239/699, Busy 0
14:13:50.791 CFG: CR 239/699, Busy 0
14:13:50.802 CFG: CR 239/699, Busy 0
14:13:50.802 CFG: CR 239/699, Busy 0
14:13:50.803 CFG: CR 248/699, Busy 0
14:13:50.814 CFG: CR 257/699, Busy 0
14:13:50.814 CFG: CR 266/699, Busy 0
14:13:50.815 CFG: CR 278/699, Busy 0
14:13:50.816 CFG: CR 279/699, Busy 0
14:13:50.826 CFG: CR 280/699, Busy 0
14:13:50.827 CFG: CR 288/699, Busy 0
14:13:50.828 CFG: CR 305/699, Busy 0
14:13:50.828 CFG: CR 309/699, Busy 0
14:13:50.839 CFG: CR 313/699, Busy 0
14:13:50.839 CFG: CR 317/699, Busy 0
14:13:50.840 CFG: CR 320/699, Busy 0
14:13:50.841 CFG: CR 322/699, Busy 0
14:13:50.851 CFG: CR 328/699, Busy 0
14:13:50.852 CFG: CR 332/699, Busy 0
14:13:50.853 CFG: CR 344/699, Busy 0
14:13:50.866 CFG: CR 359/699, Busy 0
14:13:50.867 CFG: CR 376/699, Busy 0
14:13:50.867 CFG: CR 376/699, Busy 0
14:13:50.868 CFG: CR 376/699, Busy 0
14:13:50.879 CFG: CR 376/699, Busy 0
14:13:51.020 CFG: Saved, Count 1, Bytes 4096
14:13:52.099 CFG: Saved, Count 2, Bytes 4096
14:13:52.723 APP: Restarting
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 188777542, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DOUT, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:12356
load:0x40080400,len:2720
entry 0x40080594

00:00:00.003 HDW: ESP32-U4WDH
00:00:00.029 UFS: FlashFS mounted with 304 kB free
00:00:00.078 CFG: Loaded from File, Count 2
00:00:00.087 QPC: Count 1
00:00:00.260 BRY: Berry initialized, RAM used=3961 bytes
00:00:00.331 Project tasmota - Tasmota Version 2022.1.4(tasmota)-2_0_2_2(2022-02-10T13:09:51)
00:00:01.014 WIF: WifiManager active for 3 minutes
00:00:01.794 HTP: Web server active on tasmota-A5D4DC-5340 with IP address 192.168.4.1
00:00:06.932 QPC: Reset
00:10:02.400 APP: Serial logging disabled
��?␀ets Jul 29 2019 12:21:46


```

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by barbudor** [comment, marked as answer]
> I think it is related to configuring the GPIO 4 as Button

So just don't
Use the template as it is and read about SwitchMode command to configure the switch to behave like you want

Searching in Discussions would have let you found this was already discussed a few weeks ago
https://github.com/arendst/Tasmota/discussions/14518

**Reply by barbudor** [reply]
@1Gootch1 Converting the Switch input to a BUtton has triggered the recovery mecanism because Tasmota see the button pressed for more than 40 seconds
You may have better luck by using `Button_i` instead of `Button` but still as per the other discussions, it looks like this doesn't work

**Comment by 1Gootch1** [comment]
@barbudor,

Thanks for the Hint.
I had the wrong search key words. I focused to much on the Shelly Plus 1.
I also already read the article about Switches and Buttons but I didn't remember at this point.
For me a switch is a switch and a button a button.
Seems to work now.
Sorry for wasting your time.

Regards
